src/components/AceEditor.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import bindAll from 'lodash-es/bindAll';
+import constant from 'lodash-es/constant';
+import get from 'lodash-es/get';
+import throttle from 'lodash-es/throttle';
+import noop from 'lodash-es/noop';
+
+import {EditorLocation} from '../records';
+import {createAceEditor, createAceSessionWithoutWorker} from '../util/ace';
+
+import 'brace/ext/searchbox';
+import 'brace/mode/html';
+import 'brace/mode/css';
+import 'brace/mode/javascript';
+import 'brace/theme/monokai';
+
+const RESIZE_THROTTLE = 250;
+const NORMAL_FONTSIZE = 14;
+const LARGE_FONTSIZE = 20;
+
+class AceEditor extends React.Component {
```

Comment:
I notice that you're setting `this._editor` to a DOM element, but don't initialize it in the constructor. I think it might help the clarity of the code to set it to null first, and then to a DOM element. What do you think?

Reply:
Eh this component only shows up in the diff because I renamed it; I don't think it's worth agonizing over the code therein since the goal is to delete it soon anyway.